Ruins/src/main/resources/examplesAndDocs/template_rules.txt

```diff
@@ -284,6 +284,117 @@ preserve_plants=0
+#     rule11^3*0,100,4*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule12^2*0,100,2*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule13^0,100,stone,cobblestone
+#     ^0,100,preserveBlock
+# This example generates pillars of random height from 1-5 blocks, tending to be increasingly crumbly the taller they are. The
+# selection of which variants to use for rules in a particular group is always based on the first rule of that group (i.e., the
+# rule with an = instead of ^). In this case, the first rule effectively has 5 variants due to its weight of 5, even though they
+# are all identical. As a result, a variant index of 1 to 5 will be randomly chosen for this group per structure, and every one of
+# its constituent rules will use its variant corresponding to that same index in that structure. Suppose 3 is chosen. rule9 will
+# be stone, because it's always stone. rule 10's 3rd variant is 89% stone/11% cobble, rule11's is 80% stone/20% cobble. rule12's
+# 3rd variant is preserveBlock, so this pillar is only 3 blocks high. rule13 only has 2 variants--if the group selection goes past
+# the end of one its rules (rule13 has no 3rd variant), it simply uses the last one it does have, which in this case is another
+# preserveBlock.
+# 
+# Rule Variant Group Duplication: Suppose you have a structure with four crumbly pillars, as described in the previous example.
+# You could make four stacks of rules9 through 13. Different structures would have different pillar heights, sure, but all four
+# pillars of any one structure would be of the exact same height. To get variation within the same structure, you're going to need
+# a separate group of rules for each pillar--one using rule9,10,11,12,13, one using rule14,15,16,17,18...
+#     rule14=5*0,100,stone
+#     rule15^4*0,100,8*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule16^3*0,100,4*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule17^2*0,100,2*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule18^0,100,stone,cobblestone
+#     ^0,100,preserveBlock
+# ...one using rule19,20,21,22,23--but wait! These rule groups are identical except for their names, and there's an easier way. By
+# putting a repeat count of the form "<repeats>*" in front of the first rule name in a variant group (the one with an =), that
+# many identical groups are automatically created. This is a bit tricky, because it does create new rules, which affects the way
+# rules are numbered. To continue the example, rule9 could be changed slightly to take advantage of this feature:
+#     4*rule9=5*0,100,stone
+#     rule10^4*0,100,8*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule11^3*0,100,4*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule12^2*0,100,2*stone,cobblestone
+#     ^0,100,preserveBlock
+#     rule13^0,100,stone,cobblestone
+#     ^0,100,preserveBlock
+# This creates rule9 through rule13 as before, but also creates rule14 through rule28, so you don't have to.
```

Comment:
2 characters difference creates a whole lot of "invisible" rules ... that's unfortunate

What happens when rule ID overlap?

Reply:
As always, the actual rule ID is assigned based entirely on its ordinal position in the template file--the first rule is 1, the second 2, and so on. The "names" are arbitrary and irrelevant; you can name all your rules "rule666" if you want to, and they'll still have IDs 1, 2, etc. Therefore, IDs can't possibly overlap, even if you try. I'm not particularly crazy about that, but I left it as is to maintain backward compatibility.

As for the repeat count...yeah, it's kind of hinky. I find it useful, but I wouldn't b averse to dropping that feature. It doesn't really add a new capability; you can accomplish exactly the same thing with cut-and-paste. The convenience may not be worth the potential for confusion.